synadm/cli/user.py

```diff
@@ -138,6 +139,55 @@ def deactivate(ctx, helper, user_id, gdpr_erase):
         click.echo("Abort.")
 
 
+@user.command()
+@click.argument("regex", type=str)
+@click.option(
+    "--gdpr-erase", "-e", is_flag=True, default=False,
+    help="""Marks the users as GDPR-erased. This means messages sent by the
+    users will still be visible by anyone that was in the room when these
+    messages were sent, but hidden from other users joining the room
+    afterwards.""", show_default=True)
+@click.option(
+    "--dry-run", "-n", is_flag=True, default=False,
+    help="""Do everything except deactivating users."""
+)
+@click.option(
+    "--batch-size", type=int, default=500, show_default=True,
```

Comment:
Does it make sense to set this value to same as the notice send command does? (Considering once we rewrite notice send to use your new pagination solution)

Reply:
I chose 500 cause that's a large number. It was 100 at one point, but I made it higher since you know, there could be a lot of users even on a single user homeserver (I have 7186 users on my HS).

So, in the case of 500, 15 user list requests are made. In the case of 100, 72 user list requests are made.
500 isn't a *lot*, but 100 is *even less*.

It makes more sense to have a higher batch size/pagination size in cases where computation on each user isn't a lot.

(formula for calculating amount of requests needed to paginate all users: `ceil(users/batch_size)`)

Reply:
Thanks for elaborating! So regarding "computation on each user isn't a lot" we could say that `notice send` and `user deactivate` is actually kind of the same. We send one post query to Synapse for each user. So in your case 500 times in a row `deactivate` and in the other command 100 times `send notice`. I don't really know what the computational difference for a Synapse server would be. Maybe we set the `notice` command to 500 as well? What do you think?

And thanks for that formula, I now get how much computation is required for "pagination", so actually it's better to not set too tiny pages.

Hope I got all that right! :sweat_smile: 

So because of streamlining usability: What remains is that here we use `--batch-size` and over there at `notice? command we use `--paginate`. I really would like to find use the same name for that. I don't have a hard preference which one it should be as long as we streamline it. What do you think?